src/ess/livedata/dashboard/config_store.py

```diff
@@ -68,3 +76,262 @@ def _evict_oldest(self) -> None:
         oldest_keys = list(self.data.keys())[:num_to_remove]
         for key in oldest_keys:
             del self.data[key]
+
+
+class FileBackedConfigStore(UserDict[WorkflowId, dict[str, Any]]):
+    """
+    File-backed implementation of ConfigStore with optional LRU eviction.
+
+    Persists configurations to a YAML file on disk, with the same LRU eviction
+    policy as InMemoryConfigStore. Configurations are loaded from the file on
+    initialization and saved after each modification.
+
+    This implementation uses file locking (fcntl) to ensure thread-safety when
+    multiple processes access the same config file.
+
+    Parameters
+    ----------
+    file_path:
+        Path to the YAML file for storing configurations. Parent directory
+        will be created if it doesn't exist.
+    max_configs:
+        Maximum number of configurations to keep. When exceeded, oldest
+        configurations are automatically evicted. If None, no limit is enforced.
+    cleanup_fraction:
+        Fraction of configs to remove when limit exceeded (default 0.2 = 20%).
+    """
+
+    def __init__(
+        self,
+        file_path: Path | str,
+        max_configs: int | None = None,
+        cleanup_fraction: float = 0.2,
+    ) -> None:
+        super().__init__()
+        self._file_path = Path(file_path)
+        self._max_configs = max_configs
+        self._cleanup_fraction = cleanup_fraction
+
+        # Ensure parent directory exists
+        self._file_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Load existing configs from file
+        self._load_from_file()
+
+    def __setitem__(self, key: WorkflowId, value: dict[str, Any]) -> None:
+        """Save configuration with automatic LRU eviction and file persistence."""
+        super().__setitem__(key, value)
+
+        # Automatic LRU eviction if limit exceeded
+        if self._max_configs and len(self.data) > self._max_configs:
+            self._evict_oldest()
+
+        # Persist to file after modification
+        self._save_to_file()
+
+    def __delitem__(self, key: WorkflowId) -> None:
+        """Delete configuration and persist to file."""
+        super().__delitem__(key)
+        self._save_to_file()
+
+    def _evict_oldest(self) -> None:
+        """Remove oldest configs based on cleanup fraction."""
+        num_to_remove = max(1, int(len(self.data) * self._cleanup_fraction))
+        # Dict maintains insertion order in Python 3.7+
+        oldest_keys = list(self.data.keys())[:num_to_remove]
+        for key in oldest_keys:
+            del self.data[key]
```

Comment:
These operations are not atomic. Is there a risk that multiple threads will access the config at the same time?